Content.Shared/Kitchen/RecipeManager.cs

```diff
@@ -24,6 +24,27 @@ public void Initialize()
 
             Recipes.Sort(new RecipeComparer());
         }
+        /// <summary>
+        /// Check if a given id appears in any of the recipes that exist.
+        /// </summary>
+        /// <param name="solidIds"></param>
+        /// <returns></returns>
+        public bool SolidAppears(string[] solidIds)
+        {
+            for(var i = 0; i < Recipes.Count; i++)
+            {
+                for(var j = 0; j < solidIds.Length; j++)
+                {
+                    if(Recipes[i].IngredientsSolids.ContainsKey(solidIds[j]))
+                    {
+                        return true;
```

Comment:
Also, this returns `true` for the first success - so it doesn't check that all the solids in the list appear in the same recipe, or even any recipe.